---
title: "Data Info"
output:
  html_document:
    df_print: paged
---

# Literature Source

Beissinger TM, Hirsch CN, Vaillancourt B, et al. A genome-wide scan for evidence of selection in a maize population under long-term artificial selection for ear number. Genetics. 2014 Mar;196(3):829-840.

## Experimental Design

This study conducted a genome-wide detection of the selection in the *Golden Glow* maize population. This population has been through selection for increased ear number over 30 generations. 48 plants were randomly chosen from cycle 0 and cycle 30. The effective population size ($N_e$) ranged between 384 and 667 individuals.

## Material and Method

### Germplasm

-   The *Golden Glow* maize population has been through selection for increased number of ears since 1971.
-   Selection intensity is initially at 2.5–5% in the first 12 cycle and later strengthened to 0.5–1% since the 13th cycle.

### Sequencing Method and Genome Size

-   Whole-genome sequencing was performed on 48 plants randomly chosen from each cycle 0 and cycle 30, by using **Illumina** .
-   A total of 555,078,520 read pairs were obtained from eight sequencing lanes for cycle 0, 652,901,808 read pairs from nine sequencing lanes for cycle 30.

### Ancestral Species

-   For this study, cycle 0 can be treated as ancestral species, and after 30 generations, cycle 30 can be treated as domesticated population.
-   The reference genome used was *B73* Version 2.

# Download Data

```{bash}
#!/bin/sh
#SBATCH --ntasks-per-node=4
#SBATCH --nodes=1
#SBATCH --mem=16gb
#SBATCH --time=36:00:00
#SBATCH --job-name=down30
#SBATCH --mail-user=nathanchu@huskers.unl.edu
#SBATCH --mail-type=ALL
#SBATCH --error=../log/30.err
#SBATCH --output=../log/30.out

module load SRAtoolkit/2.11

# cycle 0 and cycle 30

DATA_DIR30=../data/cycle_30/sra
FASTQ_DIR30=../data/cycle_30/fastq
DATA_DIR0=../data/cycle_0/sra
FASTQ_DIR0=../data/cycle_0/fastq

SRR_IDS30=("SRR801187" "SRR801188" "SRR801189" "SRR801190"
         "SRR801191" "SRR801192" "SRR801194" "SRR801195" "SRR801196")

prefetch --max-size 200G --output-directory $DATA_DIR30 ${SRR_IDS30[@]}

for SRR_ID in ${SRR_IDS30[@]}; do
    fasterq-dump --split-files --threads 16 $DATA_DIR30/$SRR_ID30 -O $FASTQ_DIR30 &
done

SRR_IDS0=("SRR801164" "SRR801169" "SRR801170" "SRR801174"
         "SRR801175" "SRR801176" "SRR801177" "SRR801178")

prefetch --max-size 200G --output-directory $DATA_DIR0 ${SRR_IDS0[@]}

for SRR_ID in ${SRR_IDS0[@]}; do
    fasterq-dump --split-files --threads 16 $DATA_DIR0/$SRR_ID0 -O $FASTQ_DIR0 &
done

## ref seq
cd ../data/ref
wget -c https://download.maizegdb.org/B73_RefGen_v2/B73_RefGen_v2.fa.gz
gunzip B73_RefGen_v2.fa.gz
## red annotation
wget -c https://download.maizegdb.org/B73_RefGen_v2/ZmB73_5a.59_WGS.gff3.gz
gunzip ZmB73_5a.59_WGS.gff3.gz
```
